app/workers/RulesConceptsActivity/__init__.py

```diff
@@ -262,40 +266,50 @@ def _handle_table(
 
     Args:
         - table (ScanReportTable): Table object to create for.
-        - vocab (Dict[str, Dict[str, str]]): Vocab dictionary.
+        - vocab_mappings (Dict[str, Dict[str, str]]): List of vocabulary mappings containing:
+            - sr_field_id: The ID of the field in the scan report.
+            - field_data_type: The data type of the field.
+            - vocabulary_id: The ID of the vocabulary.
 
     Returns:
         - None
     """
     table_values = db.get_scan_report_values(table.pk)
     table_fields = db.get_scan_report_fields(table.pk)
 
-    # Add vocab id to each entry from the vocab dict
-    helpers.add_vocabulary_id_to_entries(table_values, vocab, table.name)
+    # Convert mappings to {field_id: vocab_id}
+    field_vocab_map = {
+        mapping["sr_field_id"]: mapping["vocabulary_id"] for mapping in vocab_mappings
+    }
+
+    # Assign vocabulary IDs
+    for value in table_values:
+        field_id = value["scan_report_field"]["id"]
+        value["vocabulary_id"] = field_vocab_map.get(field_id)
```

Comment:
```suggestion
        value["vocabulary_id"] = field_vocab_map.get(int(field_id))
```
to solve the "error" from MyPy